projects/plugins/jetpack/modules/widgets/gravatar-profile.php

```diff
@@ -217,6 +217,7 @@ public function display_personal_links( $personal_links = array() ) {
 			<ul class="grofile-urls grofile-links">
 
 			<?php foreach ( $personal_links as $personal_link ) : ?>
+				<?php if ( is_array( $personal_link ) ) : ?>
```

Comment:
Might be even more precise?
```suggestion
				<?php if ( ! empty( $personal_link['value'] ) ) : ?>
```

Reply:
Hmm, could be... the problem is, the warning comes up because the value is a `bool`. And while I'm like 99.99% sure it's always going to be `false` because that almost makes sense, since I unfortunately don't know where it's coming from and what the actual possible values are I figured `is_array()` would be a little 'safer'?

Reply:
I think I misread it in the same way, as suggesting `! empty( $personal_link )` rather than `! empty( $personal_link['value'] )`. The suggestion makes sense to me once I read it correctly.

Reply:
Yeah, my suggestion would both catch the `false` and the missing `value` key, but it's such an edge case it probably doesn't matter.

Going one step further and looking in context, I'd probably personally negate the condition and use a `continue`.

I wasn't able to reproduce the issue on my end; it seems the `bool` must come from `https://secure.gravatar.com/%s.json`, but the few sample hashes I tried always gave an empty array rather than a `false`. Perhaps there's a legacy data format. 🤷